src/Arcus.Security.Providers.UserSecrets/Extensions/SecretStoreBuilderExtensions.cs

```diff
@@ -0,0 +1,122 @@
+
+            string userSecretsId = GetUserSecretsIdFromTypeAssembly(assembly);
+            return AddUserSecrets(builder, userSecretsId, reloadOnChange);
+        }
+
+        private static string GetUserSecretsIdFromTypeAssembly(Assembly assembly)
+        {
+            var attribute = assembly.GetCustomAttribute<UserSecretsIdAttribute>();
+            if (attribute is null)
+            {
+                string assemblyName = assembly.GetName().Name;
+                throw new InvalidOperationException(
+                    $"Could not find '{nameof(UserSecretsIdAttribute)}' on assembly '{assemblyName}'. "
+                    + $"Check that the project for '{assemblyName}' has set the 'UserSecretsId' build property.");
+            }
+
+            return attribute.UserSecretsId;
+        }
+
+        /// <summary>
+        /// <para>Adds the user secrets configuration source with specified user secrets ID.</para>
+        /// <para>A user secrets ID is unique value used to store and identify a collection of secrets.</para>
+        /// </summary>
+        /// <param name="builder">The builder to create the secret store.</param>
+        /// <param name="userSecretsId">The user secrets ID.</param>
+        /// <param name="reloadOnChange">Whether the configuration should be reloaded if the file changes.</param>
+        public static SecretStoreBuilder AddUserSecrets(this SecretStoreBuilder builder, string userSecretsId, bool reloadOnChange = false)
+        {
+            Guard.NotNull(builder, nameof(builder));
+            Guard.NotNullOrWhitespace(userSecretsId, nameof(userSecretsId));
+
+            string directoryPath = GetUserSecretsDirectoryPath(userSecretsId);
+            JsonConfigurationSource source = CreateJsonFileSource(reloadOnChange, directoryPath);
+
+            var provider = new JsonConfigurationProvider(source);
+            provider.Load();
+
+            return builder.AddProvider(new UserSecretsSecretProvider(provider));
+        }
+
+        private static string GetUserSecretsDirectoryPath(string usersSecretsId)
+        {
+            string secretPath = PathHelper.GetSecretsPathFromSecretsId(usersSecretsId);
+            string directoryPath = Path.GetDirectoryName(secretPath);
+            
+            return directoryPath;
+        }
+
+        private static JsonConfigurationSource CreateJsonFileSource(bool reloadOnChange, string directoryPath)
+        {
+            IFileProvider fileProvider = null;
+            if (Directory.Exists(directoryPath))
+            {
+                fileProvider = new PhysicalFileProvider(directoryPath);
+            }
+
+            var source = new JsonConfigurationSource
+            {
+                FileProvider = fileProvider,
+                Path = SecretsFileName,
+                ReloadOnChange = reloadOnChange,
+                Optional = false
+            };
+
+            source.ResolveFileProvider();
+            if (source.FileProvider == null)
```

Comment:
This extra check and default value is something I saw in the original source, so I added it here too.